src/averageAges.js

```diff
@@ -37,7 +47,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let women = people.filter(person => person.sex === 'f');
+
+  women = withChildren
+    ? women.filter(woman => people.find(child => child.mother === woman.name))
+    : women;
+
+  const ages = women.map(person => person.died - person.born);
+  const sumOfAges = ages.reduce((sum, age) => sum + age, 0);
+  const averageAge = sumOfAges / ages.length;
```

Comment:
you repeat the same functionality 2 times, let's create a separate function for calculating average value for needed array of peopel